### DISCUSSION TITLE
Issue with TLS Configuration in syslog-ng

### DISCUSSION THREAD

**Discussion by KassCroute** [syslog-ng/syslog-ng#5202] [Q&A] [answered]
Hi everyone,

I'm trying to send logs with TLS using syslog-ng and encountering some issues. To provide some context, I have a PKI server for issuing certificates, so there shouldn't be any problems with certificates. Both my server and client are using the same CA.

Here is the error message from my syslog-ng server logs:
![image](https://github.com/user-attachments/assets/507b8220-5b65-46b8-9294-b85f363edcd8)

Here is my server conf :
![image](https://github.com/user-attachments/assets/9d33a50e-d5e4-42ee-b667-dea1350d1260)
![image](https://github.com/user-attachments/assets/3a804a32-7738-4e2d-b854-3319401df2f4)
![image](https://github.com/user-attachments/assets/1f768d32-33ba-462d-ba06-36d4cbd63e95)

And here is my client conf : 
![image](https://github.com/user-attachments/assets/1e17dd1a-dbcd-401f-b5b2-15c9dfc300ab)
![image](https://github.com/user-attachments/assets/ee7bdea6-d8b9-4902-8d84-c71a817d2aaa)
![image](https://github.com/user-attachments/assets/d549fe31-f20c-46ba-80bd-f7e60b1e5a2d)

Thanks for your help.

**Comment by bazsi** [comment]
The "TLS" alert in the error message says that the SSL implementation on the _client_ encountered an internal error, so you should check the error log on the client side.

I'd remove those ssl-options() from the client first, maybe that's an incompatible combination and try to use the more intuitive ssl-version() option, e.g. `ssl-version(tls1_3)`

**Reply by KassCroute** [reply]
Hello thanks for your answer ! 
I removed the ssl-options for the moment. 
Here is the log of my client :
`Jan  5 22:16:50 pts-srv syslog-ng[1385411]: Syslog connection established; fd='50', server='AF_INET(192.168.1.210:6514)', local='AF_INET(0.0.0.0:0)'
Jan  5 22:16:50 pts-srv syslog-ng[1385411]: Certificate subject does not match configured hostname; hostname='192.168.1.210', certificate='pts-rsyslog'
Jan  5 22:16:50 pts-srv syslog-ng[1385411]: SSL error while writing stream; tls_error='error:0A000086:SSL routines::certificate verify failed', location='/etc/syslog-ng/syslog-ng.conf:91:9'
Jan  5 22:16:50 pts-srv syslog-ng[1385411]: I/O error occurred while writing; fd='50', error='Broken pipe (32)'
Jan  5 22:16:50 pts-srv syslog-ng[1385411]: Syslog connection broken; fd='50', server='AF_INET(192.168.1.210:6514)', time_reopen='60'`

What am I supposed to do ? A new certificate or put pts-rsyslog in my hosts file and replace my ip address in my syslog-ng.conf by the name of the server ?

Thanks !

**Reply by bazsi** [reply]
Using the right hostname in the config would help. If you don't have that in DNS then adding the name to /etc/hosts would resolve that too, yes.

**Reply by KassCroute** [reply, marked as answer]
That's what I did and it worked thanks :))
